### PR TITLE
Use intrinsic type keywords in type quickinfo

### PR DESCRIPTION
**Pull request by rchande**
Fixes https://github.com/dotnet/roslyn/issues/18196

### CODE REVIEW

**Comment by rchande** [comment]
Note that this could also be tweaked to obey the option that controls use of keywords vs types in the rest of the IDE. Thoughts?
@dotnet/roslyn-ide for review

**Comment by sharwell** [comment]
> Note that this could also be tweaked to obey the option that controls use of keywords vs types in the rest of the IDE.

If that's easy, it would be a nice touch. Just make sure if we're providing that as an intentional feature that there are tests for both cases.

File: src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs [hunk 1]
```diff
@@ -589,71 +589,71 @@ void M()
...
+                MainDescription("int"));
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
         public async Task TestBuiltInInt()
         {
             await TestInClassAsync(
 @"$$int i;",
-                MainDescription("struct System.Int32"));
+                MainDescription("int"));
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs]: ❗️ This particular change concerns me enough that I think we should stop to really think through more cases. If I hover over a built-in type, I would expect it to show the true underlying type name regardless of my particular preference for built-in or full type names.

**Comment by rchande** [comment]
The bug requests that we show the same text when hovering on a tuple type as when hovering on a member that has/returns that type. We could tweak the behavior further:
* Intrinsic type by itself (`int`) -> `System.Int32`
* Intrinsic type that's  type parameter (`(int, int)`, `List<int>`) -> `int`
This would make the display less noisy when the intrinsic type is a type parameter but allow curious users to determine the underlying type (if anyone does that).  @Pilchie @DustinCampbell ?

**Comment by sharwell** [comment]
* When hovering over a variable, I think we'll quickly reach agreement that the desired Quick Info text is the type as it would follow the users Type Style preferences with the exception of using `var` (it would never show `var`). In other words, show built-in type names if preferred and otherwise use the metadata type names.

* When hovering over an array type, the IDE shows the type name in its preferred form, followed by `[]`. I think tuple types should behave the same way, which means I agree with the request in #18196.

:arrow_right: So it sounds like the only place where I have a real problem with this PR is I think hovering over a built-in type name should show the metadata name. Honestly the `var` case doesn't bother me (e.g. I wouldn't have a problem if hovering over `var` showed `int` instead of `struct System.Int32`).

**Comment by CyrusNajmabadi** [comment]
I'm with Sam on this.  I actually think showing "struct Int32" is valuable in terms of explaining to users how C# works and how the keywords are just aliases.   However, this should only be on naked keyword types, and not when showing something like ```List<int>```.

**Comment by rchande** [comment]
@CyrusNajmabadi @sharwell We all agree :)

**Comment by DustinCampbell** [comment]
> @CyrusNajmabadi @sharwell We all agree :)

We do. :smile:

File: src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs (hunk 1 above)
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs]: Agree here - per mail thread.

**Pull request closed without merging**